utils.py

```diff
@@ -415,6 +415,9 @@ def write_input_data(data: dict, input_filename: Optional[str] = None) -> str:
     )
 
     if input_filename is not None:
+        if len(input_filename.split(".")) < 2:
+            input_filename = input_filename + ".txt"
```

Comment:
Should be OK to have filenames with dots in, no? This would cause an issue if there's a dot in the name. 

I quite like using `pathlib`, which can be used quite nicely here: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix, and also for joining and reading/writing files.

Reply:
Updated: https://github.com/dwave-examples/3d-bin-packing/pull/50/commits/26cc4b4b7edcd264535a64497162bf1f928dcd07